src/e3/testsuite/__init__.py

```diff
@@ -188,17 +188,26 @@ def testsuite_main(self, args: Optional[List[str]] = None) -> int:
         temp_group.add_argument(
             "-t", "--temp-dir", metavar="DIR", default=Env().tmp_dir
         )
+        temp_group.add_argument(
+            "--no-random-temp-subdir",
+            action="store_true",
```

Comment:
You might find it easier to read the code by avoiding a negative in the variable name. As we see below, you then declare a local variable called "random_subdir = not args.no_random_[...]". Instead, what you could do is use argparse "dest" parameter, default, and then set the action to "store_false". A bit more work here, but I makes the use easier, IMO.

Reply:
Good idea, thanks!